CircuitPython_101/basic_data_structures/play_note/code.py

```diff
@@ -0,0 +1,31 @@
+import time
+import board
+import pulseio
+
+C4     = 261
+C_SH_4 = 277
+D4     = 293
+D_SH_4 = 311
+E4     = 329
+F4     = 349
+F_SH_4 = 369
+G4     = 392
+G_SH_4 = 415
+A4     = 440
+A_SH_4 = 466
+B4     = 493
+
+def play_note(note):
+    if note[0] != 0:
+        pwm = pulseio.PWMOut(board.D12, duty_cycle = 0, frequency=note[0])
+        pwm.duty_cycle = 0x7FFF
```

Comment:
I'm not sure how basic you intend for this to be, but using hex here is not as beginner friendly. If this isn't meant to be a beginner guide, ignore this suggestion.

Reply:
I marked it intermediate and I'd consider it to be post-beginner. This is the first of a series I have in mind. The goal is to help folks take the next step. Now with the M4 boards it's possible to do go beyond simple scripts. This series aims to help people do that.

Specific to this case, having to specify duty cycle as an unsigned 16 bit int isn't very beginner friendly (whether in hex or decimal). It would be better done as an int between 0 & 100 (inclusive) since duty cycle is generally thought of as a percentage, and not as the comparison value for the 16 bit PWM counter. Actually supporting both would be ideal.

Reply:
I added `.fraction` to `adafruit_seesaw.PWMOut`, and we can do the same for `pulseio.PWMOut` here. It was late in the game so I didn't add it here. `.fraction` is between 0.0 and 1.0. I'll open an issue.